### ISSUE TITLE
[Feature Request] Cupy dry installation

### ISSUE THREAD

**Issue by fiarabbit** [cupy/cupy#1833]
I'm using a IDE (PyCharm) to develop my own project, using Chainer and Cupy.
The problem is that auto-completion does not work for Cupy, because it is not installed in my local laptop, which has no GPUs.

By connecting to the GPU server, the IDE auto-completion works, but it is not always available.

Is it possible to distribute a "stub" of cupy (works on CPU, if possible)?

**Comment by kmaehashi** [comment]
Official answer: you can install CUDA Toolkit to build CuPy even if you don't have GPU. (if you are on Windows, please wait for #1781 to be merged.)

Unofficial answer: There is a hidden (undocumented and thus unsupported) configure option called `--cupy-no-cuda` (to be given to `python setup.py`), which allows compiling CuPy without dependency to CUDA. (unfortunately, the built module does not function at all.)

**Comment by fiarabbit** [comment]
@kmaehashi 
Thanks for your kind and quick answer. That was the very answer which I wanted. 

I'm a Windows user, so it seems that I cannot proceed the first answer.
Therefore, I tried to install cupy via the second answer, but I failed to install due to some nccl matter.
```hoge.sh
git clone https://github.com/cupy/cupy.git
cd cupy && python setup.py instal --cupy-no-cuda
```
```
building 'cupy.cuda.nccl' extension
C:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\BIN\x86_amd64\cl.exe /c /nologo /Ox /W3 /GL /DNDEBUG /MD -DCUPY_NO_NVTX=1 -D_GLIBCXX_USE_CXX11_ABI=0 -D_FORCE_INLINES=1 -DCUPY_NO_CUDA=1 -IC:\Users\username\.python-venv\include -IC:\Users\username\AppData\Local\Programs\Python\Python37\include -IC:\Users\username\AppData\Local\Programs\Python\Python37\include "-IC:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\INCLUDE" "-IC:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\ATLMFC\INCLUDE" "-IC:\Program Files (x86)\Windows Kits\10\include\10.0.14393.0\ucrt" "-IC:\Program Files (x86)\Windows Kits\NETFXSDK\4.6.1\include\um" "-IC:\Program Files (x86)\Windows Kits\10\include\10.0.14393.0\shared" "-IC:\Program Files (x86)\Windows Kits\10\include\10.0.14393.0\um" "-IC:\Program Files (x86)\Windows Kits\10\include\10.0.14393.0\winrt" /EHsc /Tpcupy\cuda\nccl.cpp /Fobuild\temp.win-amd64-3.7\Release\cupy\cuda\nccl.obj
nccl.cpp
C:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\BIN\x86_amd64\link.exe /nologo /INCREMENTAL:NO /LTCG /DLL /MANIFEST:EMBED,ID=2 /MANIFESTUAC:NO /LIBPATH:C:\Users\username\.python-venv\libs /LIBPATH:C:\Users\username\AppData\Local\Programs\Python\Python37\libs /LIBPATH:C:\Users\username\AppData\Local\Programs\Python\Python37 /LIBPATH:C:\Users\username\.python-venv\PCbuild\amd64 "/LIBPATH:C:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\LIB\amd64" "/LIBPATH:C:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\ATLMFC\LIB\amd64" "/LIBPATH:C:\Program Files (x86)\Windows Kits\10\lib\10.0.14393.0\ucrt\x64" "/LIBPATH:C:\Program Files (x86)\Windows Kits\NETFXSDK\4.6.1\lib\um\x64" "/LIBPATH:C:\Program Files (x86)\Windows Kits\10\lib\10.0.14393.0\um\x64" /EXPORT:PyInit_nccl build\temp.win-amd64-3.7\Release\cupy\cuda\nccl.obj /OUT:build\lib.win-amd64-3.7\cupy\cuda\nccl.cp37-win_amd64.pyd /IMPLIB:build\temp.win-amd64-3.7\Release\cupy\cuda\nccl.cp37-win_amd64.lib
nccl.obj : warning LNK4197: export 'PyInit_nccl' specified multiple times; using first specification
   Creating library build\temp.win-amd64-3.7\Release\cupy\cuda\nccl.cp37-win_amd64.lib and object build\temp.win-amd64-3.7\Release\cupy\cuda\nccl.cp37-win_amd64.exp
nccl.obj : error LNK2001: unresolved external symbol ncclCommInitRank
nccl.obj : error LNK2001: unresolved external symbol ncclReduce
nccl.obj : error LNK2001: unresolved external symbol ncclGetUniqueId
nccl.obj : error LNK2001: unresolved external symbol ncclCommDestroy
nccl.obj : error LNK2001: unresolved external symbol ncclBcast
nccl.obj : error LNK2001: unresolved external symbol ncclCommUserRank
nccl.obj : error LNK2001: unresolved external symbol ncclReduceScatter
nccl.obj : error LNK2001: unresolved external symbol ncclAllGather
nccl.obj : error LNK2001: unresolved external symbol ncclAllReduce
nccl.obj : error LNK2001: unresolved external symbol ncclCommCuDevice
nccl.obj : error LNK2001: unresolved external symbol ncclGetErrorString
build\lib.win-amd64-3.7\cupy\cuda\nccl.cp37-win_amd64.pyd : fatal error LNK1120: 11 unresolved externals
error: command 'C:\\Program Files (x86)\\Microsoft Visual Studio 14.0\\VC\\BIN\\x86_amd64\\link.exe' failed with exit status 1120
```
Do I need locate the nccl library even if I do not have GPUs?

**Comment by kmaehashi** [comment]
That was unexpected, will be fixed in #1857.

**Comment by kmaehashi** [comment]
@fiarabbit #1857 has merged. Could you try it again?

**Comment by kmaehashi** [comment]
Closing as it is implemented. Please reopen if you need.

**Comment by fiarabbit** [comment]
@kmaehashi I'm very sorry that I could not reply this thread due to fever.
I personally merged #1857 and I succeeded to dry-install cupy.

I appreciate this feature! Thanks for help!

**Comment by kmaehashi** [comment]
@fiarabbit No problem, take care! 😄
